mapie/regression/utils/ccp_phi_function.py

```diff
@@ -0,0 +1,974 @@
+                        else:
+                            error_ind[param] = [i]
+                    elif not isinstance(self, (PolynomialPhiFunction,
+                                               GaussianPhiFunction)):
+                        if param in warn_ind:
+                            warn_ind[param].append(i)
+                        else:
+                            warn_ind[param] = [i]
+
+        if len(warn_ind) > 0:
+            warn_msg = ""
+            for param, inds in warn_ind.items():
+                warn_msg += (
+                    f"The functions at index ({', '.join(map(str, inds))}) "
+                    + "of the 'functions' argument, has an unknown optional "
+                    + f"argument '{param}'.\n"
+                )
+            warnings.warn(
+                "WARNING: Unknown optional arguments.\n"
+                + warn_msg +
+                "The only recognized arguments are : 'X', 'y_pred' and 'z'. "
+                "The other optional arguments will act as parameters, "
+                "as it is always their default value which will be used."
+            )
+        if len(error_ind) > 0:
+            error_msg = ""
+            for param, inds in error_ind.items():
+                error_msg += (
+                    f"The functions at index ({', '.join(map(str, inds))}) "
+                    + "of the 'functions' argument, has an unknown required "
+                    + f"argument '{param}'.\n"
+                )
+            raise ValueError(
+                "Forbidden required argument.\n"
+                f"{error_msg}"
+                "The only allowed required argument are : 'X', "
+                "'y_pred' and 'z'.\n"
+                "Note: You can use optional arguments if you want "
+                "to. They will act as parameters, as it is always "
+                "their default value which will be used."
+            )
+        return cast(NDArray, self.functions)
+
+    @abstractmethod
+    def fit(
+        self,
+        X: ArrayLike,
+    ) -> None:
+        """
+        Fit function : Set all the necessary attributes to be able to transform
+        ``(X, y_pred, z)`` into the expected transformation.
+
+        It should set all the attributes of ``fit_attributes``
+
+        Parameters
+        ----------
+        X : Optional[ArrayLike]
+            Samples
+        """
+
+    def transform(
+        self,
+        X: Optional[ArrayLike] = None,
+        y_pred: Optional[ArrayLike] = None,
+        z: Optional[ArrayLike] = None,
+        disable_marginal_guarantee: bool = False,
+    ) -> NDArray:
+        check_is_fitted(self, self.fit_attributes)
+        self._check_transform_parameters()
+
+        params_mapping = {"X": X, "y_pred": y_pred, "z": z}
+        res = []
+
+        funct_list = list(self.functions_)
+        if not disable_marginal_guarantee and self.marginal_guarantee:
+            funct_list.append(lambda X: np.ones((len(X), 1)))
```

Comment:
Change in such a way that the bias is include in the functions list (as a lambda function that return the one constant).
```suggestion
```

Reply:
Done